### PR TITLE
feat: level inside the pipe

### PR DESCRIPTION
**Pull request by aguirrekarina**
## Description

I created the secret level inside the pipe, loaded with maps and blocks.
## Changes Made

- [x] Create a new scene with the level
- [x] Create a json for the map
- [x] Create a json for the entities
- [x] Conect the level just for the test

## Screenshots (if applicable)

image

To test the implementation you have to go to the fourth pipe, when you are on top, for the moment pressing the "h" key you will enter inside and the map will appear, to return to the map you only have to approach the pipe of the secret level.

## Checklist

- [x] I have reviewed my own code changes.
- [x] I have tested these changes locally.
- [x] My code follows the project's coding style guidelines.
- [ ] I have updated the documentation accordingly.
- [x] I have added appropriate labels to this pull request.

Code Reviewers:
@OropezaHugo 
@JuanPabloArequipa

### CODE REVIEW

File: MarioGame/Source/Systems/PlayerMovementSystem.cs [hunk 1]
```diff
@@ -46,12 +48,12 @@ public override void Update(GameTime gameTime, IEnumerable<Entity> entities)
...
                     if (gameTime != null) HandleUpKey(gamePadState, keyboardState, collider, animation, movement, gameTime);
                     LimitSpeed(collider, collider.maxSpeed);
-                    CreateImvisibleWall(camera,collider);
+                    CreateImvisibleWall(camera, collider);
                 }
             }
         }
 
-        private static void CreateImvisibleWall(CameraComponent camera,ColliderComponent collider)
+        private static void CreateImvisibleWall(CameraComponent camera, ColliderComponent collider)
```
Comment (JuanPabloArequipa) [inline MarioGame/Source/Systems/PlayerMovementSystem.cs:56]: Typo: should be 'Invisible'

File: MarioGame/Source/Entities/EntityFactory.cs [hunk 2]
```diff
@@ -150,6 +150,17 @@ public static Entity CreateEntity(EntityData entityData, World physicsWorld)
                     entity.AddComponent(ductExtensionAnimationComponent);
                     entity.AddComponent(new ColliderComponent(physicsWorld, entityData.position.x, entityData.position.y, ductExtensionAnimationComponent.textureRectangle, BodyType.Static));
                     break;
+                case EntityType.DUCTCROSS:
+                    AnimationComponent ductCrossAnimationComponent = new AnimationComponent(Animations.entityTextures[entityData.name], 131, 128);
+                    entity.AddComponent(ductCrossAnimationComponent);
+                    entity.AddComponent(new ColliderComponent(physicsWorld, entityData.position.x, entityData.position.y, ductCrossAnimationComponent.textureRectangle, BodyType.Static));
+                    break;
+                case EntityType.DUCTCROSS1:
```
Comment (JuanPabloArequipa) [inline MarioGame/Source/Entities/EntityFactory.cs:158]: rename DUCTCROSS1 to a better representation of the entity

File: MarioGame/Utils/DataStructures/EntitesName.cs [hunk 3]
```diff
@@ -21,6 +21,8 @@ public enum EntitiesName
         COINBLOCK,
         DUCT,
         DUCTEXTENSION,
-        BLOCKERBLOCKBROWN
+        BLOCKERBLOCKBROWN,
+        DUCTCROSS,
+        DUCTCROSS1
```
Comment (JuanPabloArequipa) [inline MarioGame/Utils/DataStructures/EntitesName.cs:26]: same, rename DUCTCROSS1

File: MarioGame/Utils/DataStructures/EntityType.cs [hunk 4]
```diff
@@ -13,6 +13,8 @@ public enum EntityType
         DUCTEXTENSION,
         POWERUP,
         COINANIMATION,
-        WINFLAG
+        WINFLAG,
+        DUCTCROSS,
+        DUCTCROSS1
```
Comment (JuanPabloArequipa) [inline MarioGame/Utils/DataStructures/EntityType.cs:18]: same, rename DUCTCROSS1

File: MarioGame/Source/Scenes/LevelScene.cs [hunk 5]
```diff
@@ -181,6 +182,11 @@ public void Update(GameTime gameTime, SceneManager sceneManager)
             {
                 var playerPosition = playerEntity.GetComponent<ColliderComponent>().Position;
                 LoadEntitiesNearPlayer(playerPosition, LoadRadius);
+
+                if (IsPlayerAtSecretLocation(3620, 3776))
```
Comment (JuanPabloArequipa) [inline MarioGame/Source/Scenes/LevelScene.cs:186]: Fix the position limits of the ducts, at x=3776 Mario is out of the duct and he tele transports anyways
image

Comment (aguirrekarina) [reply MarioGame/Source/Scenes/LevelScene.cs:186]: That part is from another US in this case that is only implemented to test the scene loading in general.

**Review by JuanPabloArequipa — APPROVED** [review]

**Review by OropezaHugo — APPROVED** [review]

**Pull request merged**